File: mobile-client/android/app/src/main/java/com/smartresponsor/mobile/app/dashboard/DashboardRouteMap.kt
package com.smartresponsor.mobile.app.dashboard

/** Canonical Dashboard ownership and navigation map. */
class DashboardRouteMap {
    fun primarySections(): List<DashboardNavigationSection> = listOf(
        DashboardNavigationSection.CATALOG,
        DashboardNavigationSection.MESSAGE,
        DashboardNavigationSection.VENDOR,
    )

    fun entryFlows(): List<DashboardEntryFlow> = listOf(DashboardEntryFlow.AUTH)

    fun navigationOwners(): Map<String, String> = mapOf(
        "catalog" to "catalog/navigation",
        "vendor" to "vendor/navigation",
        "order" to "order/ownership",
    )
}

File: mobile-client/ios/App/Dashboard/DashboardRouteMap.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// Canonical dashboard route map.
public struct DashboardRouteMap: Sendable {
    public init() {}

    public func primarySections() -> [DashboardPrimarySection] {
        [.catalog, .message, .vendor, .auth]
    }

    public func secondaryFlows() -> [DashboardSecondaryFlow] {
        [.product]
    }

    public func navigationOwners() -> [String: String] {
        [
            "catalog": "catalog/navigation",
            "vendor": "vendor/navigation",
            "order": "order/ownership",
        ]
    }
}
